plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/AccountSelector.java

```diff
@@ -100,9 +99,26 @@ public int selectAccount(String email) {
     return index;
   }
 
+  /**
+   * Works the same way as {@code selectAccount} does, except that if there exists only one
+   * account signed in, that account will always be selected regardless of {@code email}.
+   *
+   * @see #selectAccount
+   */
+  public int selectAccountInSingleAccountSelectMode(String email) {
```

Comment:
Actually, I first added the logic into `selectAccount()` by accepting an additional parameter `singleAccountSelectMode`. The problem was that it was very difficult to write a Javadoc for `selectAccount()` in a clean way due to the complexity of all the possible cases. As such, it is not trivial to understand all the behaviors of `selectAccount()` either. I think having a separate method is better; the Javadocs for both methods are clean.